ygen/gogen.go

```diff
@@ -637,6 +655,20 @@ func (t *{{ .Receiver }}) GetOrCreate{{ .ListName }}(
 	}
 	return v
 }
+`
+
+	// goLeafGetterTemplate defines a template for a function that, for a
+	// particular leaf, generates a getter method.
+	goLeafGetterTemplate = `
+// Get{{ .Name }} retrieves the value of the leaf {{ .Name }} from the {{ .Receiver }}
+// struct. Caution should be exercised whilst using this method since it will return
+// the Go zero value if the field is explicitly unset.
+func (t *{{ .Receiver }}) Get{{ .Name }}() {{ .Type }} {
```

Comment:
Would it be worth also documenting an example pattern for the client, i.e. how to check if it is set and then get it?

This is fairly obvious to any existing proto users, but might not be so for other users.

1) To check if it is set one *doesn't* use a getter.
if foo.Bar != nil ...

2) To just read regardless, one uses the getter to prevent nil panics.
foo.GetBar()

Reply:
Good idea. I've added a couple of sentences to this comment that covers this.